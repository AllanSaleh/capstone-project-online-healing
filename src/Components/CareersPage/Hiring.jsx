import React from 'react';
import { useTranslation } from 'react-i18next';

export default function Hiring() {
  const { t } = useTranslation();

  return (
    <div className="bg-yellow-primary px-sides h-2/5 flex flex-col justify-evenly py-4">
      <div className="text-3xl lg:text-title">{t('Open.Hiring')}</div>
      <div className="text-xl lg:text-subtitle text-gray-600 mt-4 text-justify">
        {t('Open.HiringIntro')}
      </div>
    </div>
  );
}
