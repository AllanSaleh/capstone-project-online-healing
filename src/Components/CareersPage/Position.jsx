import React from 'react';

export default function Position({ title1, subtitle1, title2, subtitle2 }) {
  return (
    <div className="flex flex-col justify-between w-11/12 border-b-2 border-gray-700 mb-6 py-4">
      <div>
        <div className="text-blue-dark text-lg lg:text-subtitle">{title1}</div>
        <div className="text-gray-700 text-base lg:text-paragraph">{subtitle1}</div>
      </div>

      <div className="flex flex-col">
        <div className="text-lg lg:text-subtitle text-blue-dark">{title2}</div>
        <div className="text-gray-700 text-base lg:text-paragraph">{subtitle2}</div>
      </div>
    </div>
  );
}
