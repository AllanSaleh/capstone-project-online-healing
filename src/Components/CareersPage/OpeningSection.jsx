import React from 'react';
import { useTranslation } from 'react-i18next';

import Position from './Position';

export default function OpeningSection() {
  const { t } = useTranslation();

  const positions = [
    {
      title1: 'Senior Software Developer | Backend | Remote in the USA',
      subtitle1:
        'Healing is looking for a Senior Software Developer to join our Application Development team.',
      title2: 'Engineering',
      subtitle2: 'Application Development',
    },
    {
      title1: 'Senior Software Developer | Backend | Remote in the USA',
      subtitle1:
        'Healing is looking for a Senior Software Developer to join our Application Development team.',
      title2: 'Engineering',
      subtitle2: 'Application Development',
    },
    {
      title1: 'Senior Software Developer | Backend | Remote in the USA',
      subtitle1:
        'Healing is looking for a Senior Software Developer to join our Application Development team.',
      title2: 'Engineering',
      subtitle2: 'Application Development',
    },
    {
      title1: 'Senior Software Developer | Backend | Remote in the USA',
      subtitle1:
        'Healing is looking for a Senior Software Developer to join our Application Development team.',
      title2: 'Engineering',
      subtitle2: 'Application Development',
    },
    {
      title1: 'Senior Software Developer | Backend | Remote in the USA',
      subtitle1:
        'Healing is looking for a Senior Software Developer to join our Application Development team.',
      title2: 'Engineering',
      subtitle2: 'Application Development',
    },
  ];

  return (
    <div id="Openings" className="h-section flex flex-col justify-around px-sides py-8">
      <div>
        <div className="text-3xl lg:text-title">{t('Open.Positions')}</div>
        <div className="text-xl lg:text-subtitle text-gray-700 mt-4">{t('Open.Instruct')}</div>
        <a
          className="text-xl lg:text-subtitle text-blue-dark hover:underline"
          href="mailto:recruitment@healing.com"
        >
          recruitment@healing.com
        </a>
      </div>

      <div className="h-3/4 overflow-y-scroll my-4 lg:my-0">
        {positions.map((position) => (
          <Position
            title1={position.title1}
            subtitle1={position.subtitle1}
            title2={position.title2}
            subtitle2={position.subtitle2}
          />
        ))}
      </div>
    </div>
  );
}
