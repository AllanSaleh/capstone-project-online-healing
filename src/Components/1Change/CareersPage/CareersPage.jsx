import React from 'react';

import Hiring from './Hiring';
import OpeningSection from './OpeningSection';

const vh = Math.max(document.documentElement.clientHeight || 0, window.innerHeight || 0);

function Careers() {
  return (
    <>
      <div className="h-firstsection pt-navbar">
        <div className="px-sides flex flex-col justify-evenly h-3/5">
          <div>
            <div className="text-title">CAREERS AT HEALING</div>
            <div className="text-gray-600">Be a part of making people feel better.</div>
          </div>
          <div className="text-paragraph text-gray-600">
            Our SEO software cuts through mountains of data to surface critical insights. We build
            and maintain systems that process massive amounts of data (we're talking 36 trillion
            records per day and multiple petabytes of storage.) We model transparent and empathetic
            marketing for the world. We educate our community, making every effort to help them
            improve their skill. And we do it all by fostering a culture that encourages
            accountability, empathy, and transparency.
          </div>
          <div className="text-paragraph text-gray-600">What role will you play?</div>
          <button
            onClick={() => {
              const y =
                document.getElementById('Openings').getBoundingClientRect().top +
                window.pageYOffset -
                0.1 * vh;
              window.scrollTo({ top: y, behavior: 'smooth' });
            }}
            type="button"
            className="w-64 h-12 text-subtitle bg-blue-dark rounded-lg border-2 border-transparent hover:bg-white hover:text-blue-dark hover:border-blue-dark"
          >
            SEE OUR OPEN LIST
          </button>
        </div>
        <Hiring />
      </div>
      <OpeningSection />
    </>
  );
}

export default Careers;
