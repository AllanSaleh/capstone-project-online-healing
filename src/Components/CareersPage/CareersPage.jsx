import React from 'react';
import { useTranslation } from 'react-i18next';

import Hiring from './Hiring';
import OpeningSection from './OpeningSection';

const vh = Math.max(document.documentElement.clientHeight || 0, window.innerHeight || 0);

export default function CareersPage() {
  window.scrollTo(0, 0);
  const { t } = useTranslation();

  return (
    <div>
      <div className="lg:h-firstsection pt-navbar">
        <div className="px-sides flex flex-col justify-evenly h-3/5 py-4">
          <div>
            <div className="text-3xl lg:text-title">{t('Open.Title')}</div>
            <div className="text-xl lg:text-subtitle text-gray-600 mt-4">{t('Open.Subtitle')}</div>
          </div>
          <div className="text-lg lg:text-paragraph text-gray-600 mt-4 text-justify">{t('Open.Intro')}</div>
          <div className="self-center lg:self-auto text-lg lg:text-paragraph text-gray-600 mt-4">{t('Open.Question')}</div>
          <button
            onClick={() => {
              const y =
                document.getElementById('Openings').getBoundingClientRect().top +
                window.pageYOffset -
                0.1 * vh;
              window.scrollTo({ top: y, behavior: 'smooth' });
            }}
            type="button"
            className="uppercase transition-all duration-300 self-center lg:self-auto w-52 lg:w-64 h-12 text-xl lg:text-subtitle bg-blue-dark rounded-lg border-2 border-transparent hover:bg-white hover:text-blue-dark hover:border-blue-dark"
          >{t('Careers.Btn')}</button>
        </div>
        <Hiring />
      </div>
      <OpeningSection />
    </div>
  );
}
