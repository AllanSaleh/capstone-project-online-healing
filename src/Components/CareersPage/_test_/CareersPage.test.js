import * as React from 'react';
import * as renderer from 'react-test-renderer';
import CareersPage from '../CareersPage';

it('Careers renders correctly', () => {
  const tree = renderer.create(<CareersPage />).toJSON();
  expect(tree).toMatchSnapshot();
});
