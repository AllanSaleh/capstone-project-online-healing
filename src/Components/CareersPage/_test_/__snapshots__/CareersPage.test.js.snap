// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Careers renders correctly 1`] = `
<div>
  <div
    className="lg:h-firstsection pt-navbar"
  >
    <div
      className="px-sides flex flex-col justify-evenly h-3/5 py-4"
    >
      <div>
        <div
          className="text-3xl lg:text-title"
        >
          Open.Title
        </div>
        <div
          className="text-xl lg:text-subtitle text-gray-600 mt-4"
        >
          Open.Subtitle
        </div>
      </div>
      <div
        className="text-lg lg:text-paragraph text-gray-600 mt-4 text-justify"
      >
        Open.Intro
      </div>
      <div
        className="self-center lg:self-auto text-lg lg:text-paragraph text-gray-600 mt-4"
      >
        Open.Question
      </div>
      <button
        className="uppercase transition-all duration-300 self-center lg:self-auto w-52 lg:w-64 h-12 text-xl lg:text-subtitle bg-blue-dark rounded-lg border-2 border-transparent hover:bg-white hover:text-blue-dark hover:border-blue-dark"
        onClick={[Function]}
        type="button"
      >
        Careers.Btn
      </button>
    </div>
    <div
      className="bg-yellow-primary px-sides h-2/5 flex flex-col justify-evenly py-4"
    >
      <div
        className="text-3xl lg:text-title"
      >
        Open.Hiring
      </div>
      <div
        className="text-xl lg:text-subtitle text-gray-600 mt-4 text-justify"
      >
        Open.HiringIntro
      </div>
    </div>
  </div>
  <div
    className="h-section flex flex-col justify-around px-sides py-8"
    id="Openings"
  >
    <div>
      <div
        className="text-3xl lg:text-title"
      >
        Open.Positions
      </div>
      <div
        className="text-xl lg:text-subtitle text-gray-700"
      >
        Open.Instruct
      </div>
      <a
        className="text-xl lg:text-subtitle text-blue-dark hover:underline"
        href="mailto:recruitment@healing.com"
      >
        recruitment@healing.com
      </a>
    </div>
    <div
      className="h-3/4 overflow-y-scroll my-4 lg:my-0"
    >
      <div
        className="flex flex-col justify-between w-11/12 border-b-2 border-gray-700 mb-6 py-4"
      >
        <div>
          <div
            className="text-blue-dark text-lg lg:text-subtitle"
          >
            Senior Software Developer | Backend | Remote in the USA
          </div>
          <div
            className="text-gray-700 text-base lg:text-paragraph"
          >
            Healing is looking for a Senior Software Developer to join our Application Development team.
          </div>
        </div>
        <div
          className="flex flex-col"
        >
          <div
            className="text-lg lg:text-subtitle text-blue-dark"
          >
            Engineering
          </div>
          <div
            className="text-gray-700 text-base lg:text-paragraph"
          >
            Application Development
          </div>
        </div>
      </div>
      <div
        className="flex flex-col justify-between w-11/12 border-b-2 border-gray-700 mb-6 py-4"
      >
        <div>
          <div
            className="text-blue-dark text-lg lg:text-subtitle"
          >
            Senior Software Developer | Backend | Remote in the USA
          </div>
          <div
            className="text-gray-700 text-base lg:text-paragraph"
          >
            Healing is looking for a Senior Software Developer to join our Application Development team.
          </div>
        </div>
        <div
          className="flex flex-col"
        >
          <div
            className="text-lg lg:text-subtitle text-blue-dark"
          >
            Engineering
          </div>
          <div
            className="text-gray-700 text-base lg:text-paragraph"
          >
            Application Development
          </div>
        </div>
      </div>
      <div
        className="flex flex-col justify-between w-11/12 border-b-2 border-gray-700 mb-6 py-4"
      >
        <div>
          <div
            className="text-blue-dark text-lg lg:text-subtitle"
          >
            Senior Software Developer | Backend | Remote in the USA
          </div>
          <div
            className="text-gray-700 text-base lg:text-paragraph"
          >
            Healing is looking for a Senior Software Developer to join our Application Development team.
          </div>
        </div>
        <div
          className="flex flex-col"
        >
          <div
            className="text-lg lg:text-subtitle text-blue-dark"
          >
            Engineering
          </div>
          <div
            className="text-gray-700 text-base lg:text-paragraph"
          >
            Application Development
          </div>
        </div>
      </div>
      <div
        className="flex flex-col justify-between w-11/12 border-b-2 border-gray-700 mb-6 py-4"
      >
        <div>
          <div
            className="text-blue-dark text-lg lg:text-subtitle"
          >
            Senior Software Developer | Backend | Remote in the USA
          </div>
          <div
            className="text-gray-700 text-base lg:text-paragraph"
          >
            Healing is looking for a Senior Software Developer to join our Application Development team.
          </div>
        </div>
        <div
          className="flex flex-col"
        >
          <div
            className="text-lg lg:text-subtitle text-blue-dark"
          >
            Engineering
          </div>
          <div
            className="text-gray-700 text-base lg:text-paragraph"
          >
            Application Development
          </div>
        </div>
      </div>
      <div
        className="flex flex-col justify-between w-11/12 border-b-2 border-gray-700 mb-6 py-4"
      >
        <div>
          <div
            className="text-blue-dark text-lg lg:text-subtitle"
          >
            Senior Software Developer | Backend | Remote in the USA
          </div>
          <div
            className="text-gray-700 text-base lg:text-paragraph"
          >
            Healing is looking for a Senior Software Developer to join our Application Development team.
          </div>
        </div>
        <div
          className="flex flex-col"
        >
          <div
            className="text-lg lg:text-subtitle text-blue-dark"
          >
            Engineering
          </div>
          <div
            className="text-gray-700 text-base lg:text-paragraph"
          >
            Application Development
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
